### PR TITLE
Wc/pms a6 permissions update

### PR DESCRIPTION
**Pull request by willcrick**
Update PMS sample to use Android 6.0 (23/M) permissions model for write external storage. Involved a bit of refactoring.

### CODE REVIEW

**Comment by willcrick** [comment]
@doneill can you review?

File: picture-marker-symbols/src/main/java/com/esri/arcgisruntime/sample/picturemarkersymbols/MainActivity.java [hunk 1]
```diff
@@ -114,75 +130,122 @@ public void run() {
...
+      //save the orange marker app resource to disk
+      if (saveFileToExternalStorage()) {
+        createPictureMarkerSymbolFromFile();
+      }
+    }
+  }
+
+  @Override
+  public void onRequestPermissionsResult(int requestCode,
+      String permissions[], int[] grantResults) {
```
Comment (doneill) [inline picture-marker-symbols/src/main/java/com/esri/arcgisruntime/sample/picturemarkersymbols/MainActivity.java]: Annotate parameter `permissions` and `grantResults` as `@NonNull`

File: picture-marker-symbols/src/main/java/com/esri/arcgisruntime/sample/picturemarkersymbols/MainActivity.java [hunk 2]
```diff
@@ -114,75 +130,122 @@ public void run() {
...
+    Bitmap bm = BitmapFactory.decodeResource(getResources(), R.drawable.pin_blank_orange);
 
-      try {
-        OutputStream outStream = new FileOutputStream(file);
-        bm.compress(Bitmap.CompressFormat.PNG, 100, outStream);
-        outStream.flush();
-        outStream.close();
+    //create new ArcGIS temp folder
+    File folder = new File(mArcGISTempFolderPath);
+    folder.mkdirs();
```
Comment (doneill) [inline picture-marker-symbols/src/main/java/com/esri/arcgisruntime/sample/picturemarkersymbols/MainActivity.java]: Result is being ignored, you should test the result and log if successful and send toast message if fails.

You should do similar in `onDestroy` and log result.

**Comment by doneill** [comment]
@willcrick I made the updates, please review.

**Comment by willcrick** [comment]
@doneill looks good. Tested on 6.0 emulator and 5.0 device.

**Pull request merged**